### PR TITLE
fix #113, better comments handling in properties declarations

### LINKED ISSUE #113: Properties to generate comments are not consistent between platforms?
Can it be that the "comment" attribute is expected in different places for different platforms? In that case, maybe we could or unify it or support both (to avoid breaking someone's setup).

on Web/Scss:
```
if (prop.attributes.comment)
    to_ret_prop = to_ret_prop.concat(' // ' + prop.attributes.comment);
```
https://github.com/amzn/style-dictionary/blob/f385fc00540de7297b282b0851e45195d1b02eb2/lib/common/formats.js#L28-L30

on Android:
```
<% if (prop.comment) { %><!-- <%= prop.comment %> --><% } %>
```
https://github.com/amzn/style-dictionary/blob/c2c31cb381be05d8324838361188879cd8685f3c/lib/common/templates/android/dimens.template#L26

### PR DESCRIPTION
**Pull request by didoo**
*Issue #, if available:* #113 

*Description of changes:*
I have updated the templates and formats, so that all the comments are expected as `prop.comment`, updated the tests and finally updated also the documentation. 

Notice: I don't know if that's enough or I have missed something (for example the `xxx.h.template` and `xxx.m.template` have been left untouched). Please let me know if there something else to do/fix.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: test/formats/es6Constants.js [hunk 1]
```diff
@@ -11,42 +11,42 @@
  * and limitations under the License.
  */
 
-var assert  = require('chai').assert,
-    formats = require('../../lib/common/formats');
+// var assert  = require('chai').assert,
```
Comment (dbanksdesign) [inline test/formats/es6Constants.js]: Did you mean to comment all these lines out?

File: docs/package_structure.md [hunk 2]
```diff
@@ -110,6 +110,27 @@ You can organize and name your style properties however you want, there are no r
 
 Also, the CTI structure provides a good mechanism to target transforms for specific kinds of properties. All of the transforms provided by the framework use the CTI structure to know if it should be applied. For instance, the 'color/hex' transform only applies to properties of the category 'color'.
 
+You can also add a _comment_ to a style property:
```
Comment (dbanksdesign) [inline docs/package_structure.md]: Love that you added this to the documentation.

File: lib/common/formats.js [hunk 3]
```diff
@@ -25,7 +25,7 @@ function variablesWithPrefix(prefix, properties) {
   return _.map(properties, function(prop) {
       var to_ret_prop = prefix + prop.name + ': ' + (prop.attributes.category==='asset' ? '"'+prop.value+'"' : prop.value) + ';';
 
-      if (prop.attributes.comment)
+      if (prop.comment)
```
Comment (dbanksdesign) [inline lib/common/formats.js]: Should we add this to the javascript/es6 format on line 267 below?

Comment (didoo) [reply lib/common/formats.js]: yes we should. sorry I missed it*. done.

* re. this I'll open an issue for discussion, about templates vs formats.

File: test/formats/es6Constants.js (hunk 1 above)
Comment (didoo) [reply test/formats/es6Constants.js]: Strangely, when I pushed the first time, probably a pre-hook or something triggering an eslint check was complaining about unused variables, but now not anymore. So I have reverted the lines (that actually are not used, so in theory should be commented).

**Pull request merged**